pkg/vsphere/tasks/waiter.go

```diff
@@ -82,3 +87,29 @@ func WaitForResult(ctx context.Context, f func(context.Context) (ResultWaiter, e
 	}
 	return info, nil
 }
+
+func WaitAndRetryForResult(ctx context.Context, f func(context.Context) (ResultWaiter, error)) (*types.TaskInfo, error) {
+	rand.NewSource(time.Now().UnixNano()) //creates a more unique random
```

Comment:
I previously had an exponential backoff type solution. but in reality we should not need exponential back off here. That might even be overkill since this is not a truly permanent solution.